### PR TITLE
8309562: [JVMCI] Export symbols used by VirtualThread notifyJvmti intrinsics to JVMCI compilers.

### PR DESCRIPTION
**Pull request by mur47x111**
This PR allows JVMCI compiler to implement VirtualThread notifyJvmti intrinsics.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8309562](https://bugs.openjdk.org/browse/JDK-8309562): [JVMCI] Export symbols used by VirtualThread notifyJvmti intrinsics to JVMCI compilers. (**Enhancement** - P4)

### Reviewers
 * [Doug Simon](https://openjdk.org/census#dnsimon) (@dougxc - Committer) ⚠️ Review applies to [c52b76ec](https://git.openjdk.org/jdk/pull/14348/files/c52b76eca427615aa5c2873d13fe1ee2997f7291)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14348/head:pull/14348` \
`$ git checkout pull/14348`

Update a local copy of the PR: \
`$ git checkout pull/14348` \
`$ git pull https://git.openjdk.org/jdk.git pull/14348/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14348`

View PR using the GUI difftool: \
`$ git pr show -t 14348`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14348.diff">https://git.openjdk.org/jdk/pull/14348.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14348#issuecomment-1580168431)

### CODE REVIEW

File: src/hotspot/share/jvmci/vmStructs_jvmci.cpp [hunk 1]
```diff
@@ -215,6 +216,10 @@
   nonstatic_field(JavaThread,                  _poll_data,                                    SafepointMechanism::ThreadData)        \
   nonstatic_field(JavaThread,                  _stack_overflow_state._reserved_stack_activation, address)                            \
   nonstatic_field(JavaThread,                  _held_monitor_count,                           int64_t)                               \
+  JVMTI_ONLY(nonstatic_field(JavaThread,       _is_in_VTMS_transition,                        bool))                                 \
+  JVMTI_ONLY(nonstatic_field(JavaThread,       _is_in_tmp_VTMS_transition,                    bool))                                 \
+                                                                                                                                     \
+  static_field(JvmtiVTMSTransitionDisabler,    _VTMS_notify_jvmti_events,                     bool)                                  \
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/vmStructs_jvmci.cpp:222]: I think this should be also under `JVMTI_ONLY()`.

Comment (mur47x111) [reply src/hotspot/share/jvmci/vmStructs_jvmci.cpp]: No. Thought it was accessible without JVMTI. Thanks for pointing this out, I have pushed an update.

**Pull request closed without merging**